[quant][fx][pyper] Get first linear use of quantize_per_tensor for FQN

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #54860 [quant][fx] Add pass in convert to fold quant-dequant sequence
* **#54859 [quant][fx][pyper] Get first linear use of quantize_per_tensor for FQN**

Summary:
This is applicable to the case when a call_function linear op is one of the users of quantize op
In order to be able to map the qparams of quantize_per_tensor to the qparams of the linear operator
that consumes it, we need to use the FQN of the module with linear op for the qparmas of quantize_per_tensor.

Test Plan:
python test/test_quantization.py test_qparams_fqn
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27390505](https://our.internmc.facebook.com/intern/diff/D27390505)

supriyar committed:
[quant][fx][pyper] Get first linear use of quantize_per_tensor for FQN

Summary:
This is applicable to the case when a call_function linear op is one of the users of quantize op
In order to be able to map the qparams of quantize_per_tensor to the qparams of the linear operator
that consumes it, we need to use the FQN of the module with linear op for the qparmas of quantize_per_tensor.

Test Plan:
python test/test_quantization.py test_qparams_fqn
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx][pyper] Get first linear use of quantize_per_tensor for FQN"


Summary:
This is applicable to the case when a call_function linear op is one of the users of quantize op
In order to be able to map the qparams of quantize_per_tensor to the qparams of the linear operator
that consumes it, we need to use the FQN of the module with linear op for the qparmas of quantize_per_tensor.

Test Plan:
python test/test_quantization.py test_qparams_fqn
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27390505](https://our.internmc.facebook.com/intern/diff/D27390505)

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/fx/utils.py:
```diff
@@ -131,17 +131,25 @@ def quantize_node(quantizer, in_node, obs_module, obs_node, is_input):
+        # of linear node to get the FQN.
         users = list(obs_node.users)
-        first_use = users[0] if users else None
+        first_use_or_linear = users[0] if users else None
```
nit: maybe `first_linear_use_or_first_use`?

supriyar committed:
Update on "[quant][fx][pyper] Get first linear use of quantize_per_tensor for FQN"


Summary:
This is applicable to the case when a call_function linear op is one of the users of quantize op
In order to be able to map the qparams of quantize_per_tensor to the qparams of the linear operator
that consumes it, we need to use the FQN of the module with linear op for the qparmas of quantize_per_tensor.

Test Plan:
python test/test_quantization.py test_qparams_fqn
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27390505](https://our.internmc.facebook.com/intern/diff/D27390505)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][fx][pyper] Get first linear use of quantize_per_tensor for FQN"


Summary:
This is applicable to the case when a call_function linear op is one of the users of quantize op
In order to be able to map the qparams of quantize_per_tensor to the qparams of the linear operator
that consumes it, we need to use the FQN of the module with linear op for the qparmas of quantize_per_tensor.

Test Plan:
python test/test_quantization.py test_qparams_fqn
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27390505](https://our.internmc.facebook.com/intern/diff/D27390505)

[ghstack-poisoned]
